corgidrp/polflat.py

```diff
@@ -0,0 +1,377 @@
+            im_size (int): x-dimension of the input image (in pixels; default is size of input dataset; = 420 for the HST images)
+            N (int): Number of images to be combined for creating a matched filter (defaults to 1, may not work for N>1 right now)
+            rad_mask (float): radius in pixels used for creating a mask for band (band1=1.25, band4=1.75)
+            planet_rad (int): radius of the planet in pixels (planet_rad=50 for neptune, planet_rad=65)
+            n_pix (int): Number of pixels in radius covering the Roman CGI imaging FOV (defaults to 44 pix for Band1 HLC; 165 pixels for full shaped pupil FOV).
+            n_pad (int): Number of pixels padded with '1s'  to generate the image size 1024X1024 pixels around imaging FOV (defaults to None; rest of the FOV to reach 1024)
+            sky_annulus_rin (float): Inner radius of annulus to use for sky subtraction. In units of planet_rad. 
+                                     If both sky_annulus_rin and sky_annulus_rout = None, skips sky subtraciton.
+            sky_annulus_rout (float): Outer radius of annulus to use for sky subtraction. In units of planet_rad. 
+            image_center_x (int): x coordinate of the center pixel of the final image with two orthogonal pol components
+            image_center_y (int): y coordinate of the center pixel of the final image with two orthogonal pol components
+            separation_diameter_arcsec (float): separation between the two orthogonal pol components in arcsecs
+            alignment_angle_WP1 (int): wollaston prism angle for Pol0 - 0
+            alignment_angle_WP2 (int): wollaston prism angle for Pol45- 45
+            prism (str): wollaston prism pol0 or pol45
+            
+    	Returns:
+    		data.FlatField (corgidrp.data.FlatField): a master flat corresponding to Pol0 or Pol 45 for flat calibration using on sky images of planet in band specified
+    		
+	"""
+    
+    if im_size is None:
+        # assume square images
+        im_size = dataset[0].data.shape[0]
+    n=im_size
+    if n_pad is None:
+        n_pad = 1024 - im_size
+        if n_pad < 0:
+            n_pad = 0 
+
+    if planet is None:
+         planet=dataset[0].pri_hdr['TARGET']
+    if band is None:
+         band=dataset[0].ext_hdr['CFAMNAME']
+    
+    if planet_rad is None:
+        if planet.lower() =='neptune':
+             planet_rad = 50
+        elif planet.lower() == 'uranus':
+             planet_rad = 65
+    
+    if rad_mask is None:
+         if band[0] == "1":
+            rad_mask = 1.25
+         elif band[0] == "4":
+            rad_mask = 1.75
+
+    smooth_images=[]; raster_images_cent_pol1=[]; cent_pol1=[]; act_cents_1=[]; act_cents_2=[];raster_images_cent_pol2=[];cent_pol2=[];
+    for j in range(len(dataset)):
+        planet_image=dataset[j].data
+        prihdr=dataset[j].pri_hdr
+        exthdr=dataset[j].ext_hdr
+        planet_image_downsampled = downscale_local_mean(planet_image, (10,10))
```

Comment:
why do we need to do this? Alternatively, you could use scipy.ndimage.gaussian_filter or median_filter to smooth the image. That way we 1) don't have to worry about downsmapling and upsampling coordinates and 2) don't need to add a new package to the requirements just for this step.

Additioanlly, the amount we smooth/downsample should be a optional keyword for the function.